Makefile

```diff
@@ -20,14 +20,23 @@ third_party/make-env/conda.mk:
 
 include third_party/make-env/conda.mk
 
-build:
+.PHONY: update
+update:
 	git submodule init
 	git submodule update --init --recursive
 	pushd ${RAPIDWRIGHT_PATH} && \
 		make update_jars && \
 		popd
+
+.PHONY:
+third_party_SHA:
+	git submodule status > third_party_versions.txt
```

Comment:
There should be a note in the README or somewhere to run this target everytime a submodule is bumped, so that the GH action cache is not used and a new environment is generated.

Reply:
This target is automatically run when build target is made

Reply:
Thinking about this, we could solve it without checking in files in the codebase, but run the `git submodule status` command and output it to a file to be hashed directly in the GH action, before running the `cache` action

Reply:
That is a option and I was thinking about it. I'm not sure how we should name created file, so we wouldn't override some file in the future.

Reply:
I think something like `.submodules_status` is fine, I don't see anything that can be added that will cause issues with this

Reply:
makes sense, I'll update it shortly